### PR TITLE
Add module with common annotations processing logic

### LINKED ISSUE #556: Create module `:storio-common-annotations-processor` with common annotations processor logic

### PR DESCRIPTION
**Pull request by nikitin-da**
Closes #556 
@artem-zinnatullin Can you check please? 
Will it be better to create common annotation module? (To get rid from generics hell. And maybe it will be helpful to make common fetching logic from #557)

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
>Will it be better to create common annotation module? (To get rid from generics hell. And maybe it will be helpful to make common fetching logic from #557)

I'm afraid we can stuck with it in future if annotation processors would have different requirements at some point.

File: storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/StorIOProcessor.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+/**
+ * Base annotation processor for StorIO
+ * <p>
+ * It'll process annotations to generate StorIO Object-Mapping
+ * <p>
+ * Addition: Annotation Processor should work fast and be optimized because it's part of compilation
+ * We don't want to annoy developers, who use StorIO
+ */
+// Generate file with annotation processor declaration via another Annotation Processor!
+public abstract class StorIOProcessor
```
Comment (artem-zinnatullin) [inline storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/StorIOProcessor.java]: Nice!

File: storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/StorIOProcessor.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        if (annotatedField.getModifiers().contains(FINAL)) {
+            throw new ProcessingException(
+                    annotatedField,
+                    getColumnAnnotationClass().getSimpleName() + " can not be applied to final field: " + annotatedField.getSimpleName()
+            );
+        }
+    }
+
+    @Override
+    public synchronized void init(ProcessingEnvironment processingEnv) {
```
Comment (artem-zinnatullin) [inline storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/StorIOProcessor.java]: nullability

File: storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/StorIOProcessor.java [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        super.init(processingEnv);
+        filer = processingEnv.getFiler();
+        elementUtils = processingEnv.getElementUtils(); // why class name is "Elements" but method "getElementUtils()", OKAY..
+        messager = processingEnv.getMessager();
+    }
+
+    @Override
+    public Set<String> getSupportedAnnotationTypes() {
```
Comment (artem-zinnatullin) [inline storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/StorIOProcessor.java]: Let's make it abstract and each processor will implement it as it wants?

File: storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/StorIOProcessor.java (hunk 1 above)
Comment (artem-zinnatullin) [inline storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/StorIOProcessor.java]: Let's rename it to `StorIOAnnotationsProcessor`? Just to make it more clear for the potential readers of the code!

File: storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/generate/Common.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.pushtorefresh.storio.common.annotations.processor.generate;
+
+import com.squareup.javapoet.ClassName;
+
+public class Common {
```
Comment (artem-zinnatullin) [inline storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/generate/Common.java]: Oh yes

**Comment by nikitin-da** [comment]
Updated =)

**Comment by artem-zinnatullin** [comment]
Awesome! Thanks for doing that.

**Pull request merged**